cpu/lm4f120/periph/spi.c

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (C) 2016 Marc Poulhiès
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_lm4f120
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Marc Poulhiès <dkm@kataplop.net>
+ *
+ * @}
+ */
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "periph_conf.h"
+#include "board.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#if SPI_0_EN
+
+/**
+ * @brief Array holding one pre-initialized mutex for each SPI device
+ */
+static mutex_t locks[] =  {
+#if SPI_0_EN
+    [SPI_0] = MUTEX_INIT,
+#endif
+#if SPI_1_EN
+    [SPI_1] = MUTEX_INIT,
+#endif
+#if SPI_2_EN
+    [SPI_2] = MUTEX_INIT,
+#endif
+#if SPI_3_EN
+    [SPI_3] = MUTEX_INIT,
+#endif
+};
+
+static const struct spi_conf_s {
+    unsigned long ssi_sysctl;
+    unsigned long ssi_base;
+    unsigned long gpio_sysctl;
+    unsigned long gpio_port;
+    struct spi_pin_conf_s {
+        unsigned long clk;
+        unsigned long fss;
+        unsigned long rx;
+        unsigned long tx;
+        unsigned long mask;
+    } pins;
+} spi_confs[] = {
```

Comment:
There are some ongoing changes in the interfaces and structures for the peripheral devices. Part of it is to place configuration specific defines into the header files (periph_conf.h, periph_cpu.h) as much as possible.

As there there's no detailed specification yet, please have a look at #4780 as a reference.